lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -91,6 +92,52 @@ splitTarget(StringRef target, MLIRContext *context) {
   return {targetBase, ArrayAttr::get(context, annotationVec)};
 }
 
+/// Split out non-local paths
+SmallVector<std::tuple<std::string, std::string, std::string>>
+expandNonLocal(StringRef target) {
+  SmallVector<std::tuple<std::string, std::string, std::string>> retval;
+  StringRef circuit;
+  std::tie(circuit, target) = target.split('|');
+  while (target.count(':')) {
+    StringRef nla;
+    std::tie(nla, target) = target.split(':');
+    StringRef inst, mod;
+    std::tie(mod, inst) = nla.split('/');
+    retval.emplace_back((circuit + "|" + mod + ">" + inst).str(), mod.str(),
+                        inst.str());
+  }
+  if (target.empty()) {
+    retval.emplace_back(circuit.str(), "", "");
+  } else {
+    StringRef mod, name;
+    // remove aggregate
+    auto targetBase =
+        target.take_until([](char c) { return c == '.' || c == '['; });
+    std::tie(mod, name) = targetBase.split('>');
+    if (name.empty())
+      name = mod;
+    retval.emplace_back((circuit + "|" + target).str(), mod, name);
+  }
+  return retval;
+}
+
+/// Make an anchor
+void buildNLA(
```

Comment:
Similarly, please expand out the name and the comment header.  A year from now many of us may not remember what "NLA" means.